### ISSUE TITLE
Implement Binary Search

### ISSUE THREAD

**Issue by deutranium** [deutranium/Algorithms#41]
Implement [Binary Search](https://www.khanacademy.org/computing/computer-science/algorithms/binary-search/a/binary-search) in a language of your choice.

**Please read the contributing guidelines in CONTRIBUTING.md**

Languages claimed/implemented:
- [x] C (by @KyaBacchaHai )
- [x] Python (by @deutranium )

**Comment by Rahul30032** [comment]
Hey  I can try implementing this in CPP, if that's okay can you assign this to me?

**Comment by hackbansu** [comment]
I'm adding this in Java.

**Comment by deutranium** [comment]
@Rahul30032 @hackbansu sure

**Comment by hackbansu** [comment]
Created #45 with code in Java. Cheers :)

**Comment by Rahul30032** [comment]
after this issue was assigned to me, I found out a basic implementation of binarysearch in cpp was pushed so i implemented another problem statement mentioned in readme file in binaryseach folder i.e implementing binary search for a sorted array with duplicate elements. my output gives first and last index as well as its count.

**Comment by deutranium** [comment]
@Rahul30032 yeah it was an issue at my end, I didn't check it. Thanks for implementing a different version and good job with the code!!

**Comment by swapnil-satpathy** [comment]
Can you please assign it to me. I along with binary search will also implement lower_bound,upper_bound from scratch which are features of STL and also add functionalities for finding ceiling and floor

**Comment by deutranium** [comment]
@swapnil-satpathy sure and I'm guessing you're talking about an implementation in C.

But you should name it as `binarySearch_scratch.c` and don't remove the code present right now as it talks about the basic flow without going into intricacies.

I'll assign this to you

**Comment by codelixir** [comment]
Hey, can I implement this in golang?

**Comment by inesane** [comment]
@codelixir sure go ahead!

**Comment by royzaa** [comment]
Hi, can I implement this in dart?

**Comment by codelixir** [comment]
@royzaa yes you can go ahead
